File: event-streams/event-processor/src/main/java/com/scientiamobile/wurflmicroservice/eventprocessor/EnrichedEventData.java
package com.scientiamobile.wurflmicroservice.eventprocessor;

import com.google.gson.annotations.SerializedName;

/**
 * Event data class is the model for the data arriving from the JSON event stream. The only data which is added
 * during this phase if the timestamp (time at which JSON event is sent to output in nanoseconds).
 */
class EnrichedEventData {

    @SerializedName(value = "X-Ucbrowser-Ua")
    private String ucBrowserUa;
    @SerializedName(value = "X-Ucbrowser-Device-Ua")
    private String ucBrowserDeviceUa;
    @SerializedName(value = "Accept-Language")
    private String acceptLanguage;
    @SerializedName(value = "Accept-Encoding")
    private String acceptEncoding;
    @SerializedName(value = "Accept-Charset")
    private String acceptCharset;
    @SerializedName(value = "X-Forwarded-For")
    private String xForwardedFor;
    @SerializedName(value = "Accept")
    private String accept;
    @SerializedName(value = "User-Agent")
    private String userAgent;
    @SerializedName(value = "Video_id")
    private String videoID;
    @SerializedName(value = "Clientip")
    private String clientIP;
    @SerializedName(value = "Event")
    private String event;
    @SerializedName("X-Operamini-Features")
    private String xOperaminiFeatures;
    @SerializedName("X-Operamini-Phone")
    private String xOperaminiPhone;
    @SerializedName("X-Operamini-Phone-Ua")
    private String xOperaminiPhoneUa;
    @SerializedName("Save-Data")
    private String saveData;
    @SerializedName("X-Clacks-Overhead")
    private String xClacksOverhead;
    @SerializedName("Sec-Fetch-Site")
    private String secFetchSite;
    @SerializedName("Sec-Fetch-Mode")
    private String secFetchMode;
    @SerializedName("Request-Context")
    private String requestContext;
    @SerializedName("Correlation-Context")
    private String correlationContext;
    @SerializedName("X-Ms-Request-Root-Id")
    private String xMsRequestRootID;
    @SerializedName("X-Ms-Request-Id")
    private String xMsRequestID;

    // Device detection fields
    @SerializedName("Wurfl-Complete-Name")
    private String wurflCompleteName;
    @SerializedName("Wurfl-Device-OS")
    private String wurflDeviceOS;
    @SerializedName("Wurfl-Form-Factor")
    private String wurflFormFactor;
    @SerializedName("Wurfl-Device-Make")
    private String wurflDeviceMake;
    @SerializedName("Wurfl-Device-Model")
    private String wurflDeviceModel;

    // timestamp in nanoseconds, when the event is sent to output (not in the original stream)
    @SerializedName(value = "timestamp")
    private long timestamp;


    // -------------------------------- ACCESSOR METHODS ---------------------------------------------------------------
    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    public String getUserAgent() {
        return userAgent;
    }

    public void setUserAgent(String userAgent) {
        this.userAgent = userAgent;
    }

    public String getWurflCompleteName() {
        return wurflCompleteName;
    }

    public void setWurflCompleteName(String wurflCompleteName) {
        this.wurflCompleteName = wurflCompleteName;
    }

    public String getWurflDeviceOS() {
        return wurflDeviceOS;
    }

    public void setWurflDeviceOS(String wurflDeviceOS) {
        this.wurflDeviceOS = wurflDeviceOS;
    }

    public String getWurflFormFactor() {
        return wurflFormFactor;
    }

    public void setWurflFormFactor(String wurflFormFactor) {
        this.wurflFormFactor = wurflFormFactor;
    }

    public String getWurflDeviceMake() {
        return wurflDeviceMake;
    }

    public void setWurflDeviceMake(String wurflDeviceMake) {
        this.wurflDeviceMake = wurflDeviceMake;
    }

    public String getWurflDeviceModel() {
        return wurflDeviceModel;
    }

    public void setWurflDeviceModel(String wurflDeviceModel) {
        this.wurflDeviceModel = wurflDeviceModel;
    }
}

File: event-streams-dotnet/event-processor/EnrichedEventData.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace event_processor
{
    class EnrichedEventData
    {
        [JsonProperty("X-Ucbrowser-Ua")]
        private String ucBrowserUa;

        [JsonProperty("X-Ucbrowser-Device-Ua")]
        private String ucBrowserDeviceUa;

        [JsonProperty("Accept-Language")]
        private String acceptLanguage;

        [JsonProperty("Accept-Encoding")]
        private String acceptEncoding;

        [JsonProperty("Accept-Charset")]
        private String acceptCharset;

        [JsonProperty("X-Forwarded-For")]
        private String xForwardedFor;

        [JsonProperty("Accept")]
        private String accept;

        [JsonProperty("User-Agent")]
        private String userAgent;

        [JsonProperty("Video_id")]
        private String videoID;

        [JsonProperty("Clientip")]
        private String clientIP;

        [JsonProperty("Event")]
        private String eventResult;

        [JsonProperty("X-Operamini-Features")]
        private String xOperaminiFeatures;

        [JsonProperty("X-Operamini-Phone")]
        private String xOperaminiPhone;

        [JsonProperty("X-Operamini-Phone-Ua")]
        private String xOperaminiPhoneUa;

        [JsonProperty("Save-Data")]
        private String saveData;

        [JsonProperty("X-Clacks-Overhead")]
        private String xClacksOverhead;

        [JsonProperty("Sec-Fetch-Site")]
        private String secFetchSite;

        [JsonProperty("Sec-Fetch-Mode")]
        private String secFetchMode;

        [JsonProperty("Request-Context")]
        private String requestContext;

        [JsonProperty("Correlation-Context")]
        private String correlationContext;

        [JsonProperty("X-Ms-Request-Root-Id")]
        private String xMsRequestRootID;

        [JsonProperty("X-Ms-Request-Id")]
        private String xMsRequestID;

        // timestamp in nanoseconds, when the event is sent to output (not in the original stream)
        [JsonProperty("timestamp")]
        private long timestamp;

        public void SetTimestamp(long timestamp)
        {
            this.timestamp = timestamp;
        }

        // Device detection fields
        [JsonProperty("Wurfl-Complete-Name")]
        private String wurflCompleteName;
        [JsonProperty("Wurfl-Device-OS")]
        private String wurflDeviceOS;
        [JsonProperty("Wurfl-Form-Factor")]
        private String wurflFormFactor;
        [JsonProperty("Wurfl-Device-Make")]
        private String wurflDeviceMake;
        [JsonProperty("Wurfl-Device-Model")]
        private String wurflDeviceModel;

        // Accessor properties
        public string UserAgent { get => userAgent; set => userAgent = value; }
        public string WurflCompleteName { get => wurflCompleteName; set => wurflCompleteName = value; }
        public string WurflDeviceOS { get => wurflDeviceOS; set => wurflDeviceOS = value; }
        public string WurflDeviceMake { get => wurflDeviceMake; set => wurflDeviceMake = value; }
        public string WurflFormFactor { get => wurflFormFactor; set => wurflFormFactor = value; }
        public string WurflDeviceModel { get => wurflDeviceModel; set => wurflDeviceModel = value; }
        public string UcBrowserUa { get => ucBrowserUa; set => ucBrowserUa = value; }
        public string UcBrowserDeviceUa { get => ucBrowserDeviceUa; set => ucBrowserDeviceUa = value; }
        public string AcceptLanguage { get => acceptLanguage; set => acceptLanguage = value; }
        public string Accept { get => accept; set => accept = value; }
        public string XOperaminiPhoneUa { get => xOperaminiPhoneUa; set => xOperaminiPhoneUa = value; }
    }
}
